plugins/@grouparoo/eloqua/src/lib/export/processExportedProfiles.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  Errors,
+  ErrorWithProfileId,
+  ProcessExportedProfilesPluginMethod,
+} from "@grouparoo/core";
+import EloquaClient from "../client/client";
+import { connect } from "../connect";
+
+import { getListId, getContact, invalidate } from "./cachedMethods";
+import { GrouparooCLI } from "@grouparoo/core/dist/modules/cli";
+import error = GrouparooCLI.logger.error;
+let client: EloquaClient;
+
+export const processExportedProfiles: ProcessExportedProfilesPluginMethod =
+  async ({
+    appId,
+    appOptions,
+    remoteKey,
+    exports: _exports,
+    syncOperations,
+  }) => {
+    client = await connect(appOptions);
+    const sync = await client.bulk.checkSync(remoteKey);
+    if (sync) {
+      if (sync.status === "error") {
+        const logsResponse = await client.bulk.getSyncLogs(remoteKey);
+        let errorsMessages = [];
+        if (logsResponse.items.length > 0) {
+          for (const log of logsResponse.items) {
+            if (log["severity"] === "error") {
+              errorsMessages.push(log["message"]);
+            }
+          }
+        }
+        throw new Error(
+          `Something went wrong. Here are the possible causes: ${errorsMessages.join(
+            " - "
+          )}`
+        );
+      } else if (sync.status === "pending") {
+        return {
+          success: true,
+          processExports: {
+            profileIds: _exports.map((e) => e.profileId),
+            remoteKey,
+            processDelay: 1000,
+          },
+        };
+      } else if (sync.status === "success" || sync.status === "warning") {
+        let success = true;
+        const errors = [];
+        if (sync.status === "warning") {
+          const rejectsResponse = await client.bulk.getSyncRejects(remoteKey);
+          for (const item of rejectsResponse.items) {
+            const rejected = <ErrorWithProfileId>new Error(item["message"]);
+            rejected.profileId = getContactProfileId({
+              email: item["fieldValues"].emailAddress,
+              exports: _exports,
+            });
+            rejected.errorLevel = "info";
+            errors.push(rejected);
+            success = false;
+          }
+        }
+        await postFKChange({
+          syncOperations,
+          appId,
+          exports: _exports,
+          errors,
+        });
+        await handleLists({ appId, appOptions, exports: _exports, errors });
+        return {
+          success,
+          errors,
+        };
+      } else {
+        return {
+          success: false,
+        };
+      }
+    } else {
+      return {
+        success: true,
+        processExports: {
+          profileIds: _exports.map((e) => e.profileId),
+          remoteKey,
+          processDelay: 1000,
+        },
+      };
+    }
+  };
+
+async function postFKChange({
```

Comment:
The reason why I did it on processExportedProfiles was exactly to avoid the case in which we've deleted the old one and the new one has not been created.